pkgs/applications/window-managers/i3/default.nix

```diff
@@ -1,40 +1,98 @@
-{ fetchurl, lib, stdenv, pkg-config, makeWrapper, meson, ninja, installShellFiles, libxcb, xcbutilkeysyms
-, xcbutil, xcbutilwm, xcbutilxrm, libstartup_notification, libX11, pcre2, libev
-, yajl, xcb-util-cursor, perl, pango, perlPackages, libxkbcommon
-, xorgserver, xvfb-run, xdotool, xorg, which
-, asciidoc, xmlto, docbook_xml_dtd_45, docbook_xsl, findXMLCatalogs
-, nixosTests
+{
+  fetchurl,
+  lib,
+  stdenv,
+  pkg-config,
+  makeWrapper,
+  meson,
+  ninja,
+  installShellFiles,
+  libxcb,
+  xcbutilkeysyms,
+  xcbutil,
+  xcbutilwm,
+  xcbutilxrm,
+  libstartup_notification,
+  libX11,
+  pcre2,
+  libev,
+  yajl,
+  xcb-util-cursor,
+  perl,
+  pango,
+  perlPackages,
+  libxkbcommon,
+  xorgserver,
+  xvfb-run,
+  xdotool,
+  xorg,
+  which,
+  asciidoc,
+  xmlto,
+  docbook_xml_dtd_45,
+  docbook_xsl,
+  findXMLCatalogs,
+  nixosTests,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "i3";
-  version = "4.23";
+  version = "4.24";
 
   src = fetchurl {
-    url = "https://i3wm.org/downloads/${pname}-${version}.tar.xz";
```

Comment:
I think we fetch the tarball from the website because i3 was using autotools in the past. It has migrated to meson, so we probably can now. I'll check.